components/event-gateway/internal/events/bus/send.go

```diff
@@ -0,0 +1,98 @@
+package bus
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/kyma-project/kyma/components/event-gateway/internal/events/api"
+	"github.com/kyma-project/kyma/components/event-gateway/internal/httpconsts"
+	"github.com/kyma-project/kyma/components/event-gateway/internal/httptools"
+)
+
+var (
+	httpClientProvider  = httptools.DefaultHttpClientProvider
+	httpRequestProvider = httptools.DefaultHttpRequestProvider
+)
+
+func InitEventSender(clientProvider httptools.HttpClientProvider, requestProvider httptools.HttpRequestProvider) {
+	httpClientProvider = clientProvider
+	httpRequestProvider = requestProvider
+}
+
+// SendEvent sends the incoming request to the Sender
+func SendEvent(req *api.SendEventParameters, traceHeaders *map[string]string) (*api.SendEventResponse, error) {
+	body := new(bytes.Buffer)
+	json.NewEncoder(body).Encode(req)
+	httpReq, err := httpRequestProvider(http.MethodPost, "", body)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := make(http.Header)
+	headers.Set(httpconsts.HeaderContentType, httpconsts.ContentTypeApplicationJson)
+	httpReq.Header = headers
+
+	reqURL, err := url.ParseRequestURI(eventsTargetURL)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
If "err != nil" an error should be returned.
"reqUrl" could be nil

Reply:
This is a code copy from the `gateway` component.
However, this is fixed because it is minor.